File: src/#120/solution.go
package solution

func minimumTotal(triangle [][]int) int {
	if len(triangle) == 0 {
		return 0
	}

	for i := len(triangle) - 2; i >= 0; i-- {
		for j := range triangle[i] {
			if triangle[i+1][j+1] > triangle[i+1][j] {
				triangle[i][j] += triangle[i+1][j]
			} else {
				triangle[i][j] += triangle[i+1][j+1]
			}
		}
	}
	return triangle[0][0]
}

File: src/#120/solution.js
var minimumTotal = function (triangle) {
  if (triangle.length === 0) {
    return 0
  } else if (triangle.length === 1) {
    return triangle[0][0]
  }
  for (let i = triangle.length - 2; i >= 0; i--) {
    for (let j = 0; j < triangle[i].length; j++) {
      triangle[i][j] = triangle[i][j] + Math.min(triangle[i + 1][j], triangle[i + 1][j + 1])
    }
  }
  return triangle[0][0]
}
